### ISSUE TITLE
Practical difficulties in handling hierarchical tree of transformations

### ISSUE THREAD

**Issue by h-fehlis** [SMPTE/ris-osvp-metadata-camdkit#135]
I'm usually working in the tracking (sender) side but I have been told from people trying to implement OpenTrackIO on the graphics (receiver) side that it is quite painful to implement handling the hierarchical tree of transformations properly. To give you an example: Using the notation `T(parent, id)` with "_" meaning no parent (top-level):
this transforms list `[T(_, A), T(_, B), T(_, C), T(B, D), T(B, E), T(_, F), T(D, G)]`
must be sorted to:` [T(_, A), T(_, B), T(B, D), T(D, G), T(B, E), T(_, C), T(_, F)]`
as it represents this transformation tree:
```
    A--B-------C--F
       \-D---E
         \-G
```
and then these sorted transformations need to be accumulated. This sorting needs to be done for every incoming data packet (because the structure can change at any time), so 60 times or more per second, potentially requiring numerous string comparisons. In addition, every packet needs to be checked for circular (infinite) loops where, for example, A is the parent of B and B it the parent of C but C is also the parent of A.
On the other hand, transmitting the same structure as a list where each entry can have a sub-list of children would not require any sorting at all and no possibility for circular loops while retaining the possibility to "toggle" each sub-tree on and off. Using the notation `T(id, children)` or just `T(id)` if there are no children, the same transformation tree as above would become:
`[T(A), T(B, [T(D, [T(G)]), T(E)]), T(C), T(F)]`
In other words (leaving out the "translation" and "rotation" parts) the above list of transformations:
```
"transforms": [
  { "id": "A" },
  { "id": "B" },
  { "id": "C" },
  { "id": "D", "parentID": "B" },
  { "id": "E", "parentID": "B" },
  { "id": "F" },
  { "id": "G", "parentID": "D" }
]
```
would become:
```
"transforms": [
  { "id": "A" },
  { "id": "B", "children": [
      { "id": "D", "children": [
          { "id": "G" }
        ]
      },
      { "id": "E" }
    ]
  },
  { "id": "C" },
  { "id": "F" }
]
```
As I said, this (or something similar) would not just make the guys happy who have to implement the receiving side, it would also be much less error-prone and faster both to implement and to run. It has really just popped up when trying to implement our proposal...

**Comment by jamesmosys** [comment]
Addressed on juren-params
